### ISSUE TITLE
Single image overfitting

### ISSUE THREAD

**Issue by kopalja** [ultralytics/ultralytics#8151]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, we are training Yolov8 for pose detection task. We have simplified the task to maximum possible scenario. Our goals is to just overfit one image. 

**Training setting**
Our training dataset contains just a single image which is copied 300 times with the same annotation. Augumentations are turned-off.

CLI command:
```
yolo pose train data=dataset.yaml model=yolov8n-pose.yaml epochs=200 imgsz=640 patience=200 hsv_h=0 hsv_s=0 hsv_v=0 translate=0 scale=0 fliplr=0 mosaic=0 crop_fraction=0 erasing=0 flipud=0 seed=42
```

**Observed results**

1. Training `pose-loss` and validation `pose-loss` differs dramatically:
![Obrázok](https://github.com/ultralytics/ultralytics/assets/32540156/4a617ac6-a97f-4be9-acee-da8d8cf572f9)

2. After visual inspectoin, `best.pt` learns bboxes well, however it fails to learn keypoint positions (keypoint visualization is tiny) :
![Obrázok](https://github.com/ultralytics/ultralytics/assets/32540156/15b9da8a-3205-464d-ad24-2b76b0ad3a5d)

**Additional information**

1. Input batch visualization
 
![train_batch0](https://github.com/ultralytics/ultralytics/assets/32540156/f285427b-525f-4575-8bb5-b75d494a8f1d)

**Comment by glenn-jocher** [comment]
@kopalja hello! Thanks for reaching out with your question. 🌟

From what you've described, it seems like you're on the right track with trying to overfit a single image for your pose detection task. However, the discrepancy between training and validation loss, as well as the keypoints not being learned as expected, suggests a few things we might need to look into.

Firstly, ensure that your single image and annotations are absolutely correct and that the keypoints are well-defined. Any small error in the annotation could be amplified when the model tries to learn from this single example.

Secondly, consider reducing the complexity of your model or increasing the learning rate since you're working with a very small dataset (essentially one image). This might help the model to pick up the keypoints more effectively.

Lastly, you might want to visualize the predictions after each epoch to see how the model is learning progressively. This could give you insights into what's going wrong.

Here's a small snippet to visualize predictions after each epoch:

```python
from ultralytics import YOLO

# Load the model
model = YOLO('path/to/best.pt')

# Run prediction
results = model('path/to/your/image.jpg')

# Plot results
results[0].plot(save=True, show=True)
```

If you continue to face issues, please provide the `dataset.yaml` and `yolov8n-pose.yaml` files so we can take a closer look. Keep experimenting, and don't hesitate to ask if you have more questions!

**Comment by glenn-jocher** [comment]
@kopalja also keep in mind that 300 iterations is extremely few to expect any sort of convergence on deep learning models, try upping to 3000 or 30000 and examine your results then.
